internal/runtime/crio/crio.go

```diff
@@ -0,0 +1,178 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package crio
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/pelletier/go-toml"
+	"k8s.io/klog/v2"
+
+	"github.com/NVIDIA/k8s-kata-manager/internal/runtime"
+)
+
+// Config represents the crio config
+type Config struct {
+	*toml.Tree
+	RuntimeType           string
+	UseDefaultRuntimeName bool
+	PodAnnotations        []string
+	Path                  string
+}
+
+func Setup(o *runtime.Options) (runtime.Runtime, error) {
+	ctrdConfig, err := New(
+		WithPath(o.Path),
+		WithPodAnnotations(o.PodAnnotations...),
+		WithRuntimeType(o.RuntimeType),
+	)
+	return ctrdConfig, err
+}
+
+// New creates a crio config with the specified options
+func New(opts ...Option) (*Config, error) {
+	b := &builder{}
+	for _, opt := range opts {
+		opt(b)
+	}
+
+	return b.build()
+}
+
+// AddRuntime adds a runtime to the crio config
+func (c *Config) AddRuntime(name string, path string, setAsDefault bool) error {
+	if c == nil {
+		return fmt.Errorf("config is nil")
+	}
+
+	config := *c.Tree
+
+	// By default we extract the runtime options from the runc settings; if this does not exist we get the options from the default runtime specified in the config.
+	runtimeNamesForConfig := []string{"runc"}
```

Comment:
Something to keep in mind is that the latest crio version always have `crun` as the default runtime and not `runc`. 

cc @elezar

Reply:
Updated to be `crun`

Reply:
How will this affect runc vs crun ? We need to support "older" kube releases. crio release == kube release.